ls1intum/Artemis
# `Development`: Fix flaky ExerciseGroupIntegrationTest

### Checklist
#### General
#### Server

### Motivation and Context
The test `importExerciseGroup_successfulIntoDifferentExam` was flaky on Postgres. 

### Description
This was actually not a wrong test, but a bug in the production code where we relied on the order of the database response.
We expected to get a list of exercise groups in the order of which they are defined in the exam, but no `ORDER BY` section was sepecifed.

We now use the `INDEX()` method to sort the results based on the order defined in the exam. See https://en.wikibooks.org/wiki/Java_Persistence/JPQL#Special_Operators for more details.

I used this opportunity to clean up the `ExerciseGroupRepository`. The method `retrieveCourseOverExerciseGroup` was unused and `findByExamId` has the same issue of not ensuring the order of the list. This method was only used in tests, so I removed it.

### Steps for Testing
You could try to test this PR on a Postgress testserver. Import exercise groups from one exam into the other and check that the order of the groups is correct. However, reviewing the code and verifying that the test is successful now should be enough.

### Review Progress
#### Code Review
